xbmc/URL.cpp

```diff
@@ -547,17 +547,39 @@ std::string CURL::Get() const
   std::string strURL;
   strURL.reserve(sizeneed);
 
-  strURL = GetWithoutFilename();
-  strURL += m_strFileName;
+  strURL = GetWithoutOptions();
 
   if( !m_strOptions.empty() )
     strURL += m_strOptions;
+
   if (!m_strProtocolOptions.empty())
     strURL += "|"+m_strProtocolOptions;
 
   return strURL;
 }
 
+std::string CURL::GetWithoutOptions() const
+{
+  unsigned int sizeneed = m_strProtocol.length()
+                        + m_strDomain.length()
+                        + m_strUserName.length()
+                        + m_strPassword.length()
+                        + m_strHostName.length()
+                        + m_strFileName.length()
+                        + 10;
+
+  if (m_strProtocol.empty())
+    return m_strFileName;
+
+  std::string strURL;
+  strURL.reserve(sizeneed);
+
+  strURL = GetWithoutFilename();
+  strURL += m_strFileName;
+
+  return strURL;
+}
+
```

Comment:
I think your right but note that this is just a c&p of what was already there. Perhaps there's a reason it was done this way?